iot-edge/EdgeSolution/modules/CSharpModule/Program.cs

```diff
@@ -0,0 +1,172 @@
+        {
+            try
+            {
+                Console.WriteLine("Desired property change:");
+                Console.WriteLine(JsonConvert.SerializeObject(desiredProperties));
+
+                if (desiredProperties["TemperatureThreshold"] != null)
+                    temperatureThreshold = desiredProperties["TemperatureThreshold"];
+
+            }
+            catch (AggregateException ex)
+            {
+                foreach (Exception exception in ex.InnerExceptions)
+                {
+                    Console.WriteLine();
+                    Console.WriteLine("Error when receiving desired property: {0}", exception);
+                }
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine();
+                Console.WriteLine("Error when receiving desired property: {0}", ex.Message);
+            }
+            return Task.CompletedTask;
+        }
+
+        /// <summary>
+        /// This method is called whenever the module is sent a message from the EdgeHub. 
+        /// </summary>
+        static async Task<MessageResponse> FilterMessages(Message message, object userContext)
+        {
+            var counterValue = Interlocked.Increment(ref counter);
+            try
+            {
+                ModuleClient moduleClient = (ModuleClient)userContext;
+                var messageBytes = message.GetBytes();
+                var messageString = Encoding.UTF8.GetString(messageBytes);
+                Console.WriteLine($"Received message {counterValue}: [{messageString}]");
+
+                // Get the message body.
+                var messageBody = JsonConvert.DeserializeObject<MessageBody>(messageString);
+
+                if (messageBody != null && messageBody.machine.temperature > temperatureThreshold)
+                {
+                    Console.WriteLine($"Machine temperature {messageBody.machine.temperature} " +
+                        $"exceeds threshold {temperatureThreshold}");
+                    var filteredMessage = new Message(messageBytes);
+                    foreach (KeyValuePair<string, string> prop in message.Properties)
+                    {
+                        filteredMessage.Properties.Add(prop.Key, prop.Value);
+                    }
+
+                    filteredMessage.Properties.Add("MessageType", "Alert");
+                    await moduleClient.SendEventAsync("output1", filteredMessage);
+                }
+
+                // Indicate that the message treatment is completed.
+                return MessageResponse.Completed;
+            }
+            catch (AggregateException ex)
+            {
+                foreach (Exception exception in ex.InnerExceptions)
+                {
+                    Console.WriteLine();
+                    Console.WriteLine("Error in sample: {0}", exception);
+                }
+                // Indicate that the message treatment is not completed.
```

Comment:
I don't understand the comment. (What does "treatment" mean here"?) It's also out of sync with what the code does below: abandons the message.